server/Arcadia.Assistant.CSP/Vacations/CspEmployeeVacationsRegistry.cs

```diff
@@ -0,0 +1,259 @@
+                                    return new GetCalendarEvent.Response.Found(result.CalendarEvent);
+                                }
+
+                                return new GetCalendarEvent.Response.NotFound();
+                            },
+                            failure: err => new GetCalendarEvent.Response.NotFound());
+                    break;
+
+                case GetCalendarEventApprovals msg:
+                    this.GetVacation(msg.Event.EventId)
+                        .PipeTo(
+                            this.Sender,
+                            success: result =>
+                            {
+                                if (result != null)
+                                {
+                                    return new GetCalendarEventApprovals.SuccessResponse(result.Approvals.ToList());
+                                }
+
+                                return new GetCalendarEventApprovals.ErrorResponse($"Vacation with id {msg.Event.EventId} is not found");
+                            },
+                            failure: err => new GetCalendarEventApprovals.ErrorResponse(err.Message));
+                    break;
+
+                case InsertVacation msg:
+                    this.vacationsSyncExecutor.InsertVacation(msg.Event, msg.Timestamp, msg.CreatedBy)
+                        .PipeTo(
+                            this.Sender,
+                            success: result => new InsertVacation.Success(result.CalendarEvent, msg.CreatedBy, msg.Timestamp),
+                            failure: err => new InsertVacation.Error(err));
+                    break;
+
+                case UpdateVacation msg:
+                    this.vacationsSyncExecutor.UpdateVacation(msg.NewEvent, msg.Timestamp, msg.UpdatedBy)
+                        .PipeTo(
+                            this.Sender,
+                            success: result => new UpdateVacation.Success(result.CalendarEvent, msg.OldEvent, msg.UpdatedBy, msg.Timestamp),
+                            failure: err => new UpdateVacation.Error(err));
+                    break;
+
+                case ApproveVacation msg:
+                    this.ApproveVacation(msg)
+                        .PipeTo(
+                            this.Sender,
+                            success: result =>
+                            {
+                                if (result != null)
+                                {
+                                    return new ApproveVacation.Success(result.CalendarEvent, result.Approvals.ToList());
+                                }
+
+                                return Calendar.Abstractions.EmployeeVacations.ApproveVacation.Success.Instance;
+                            },
+                            failure: err => new ApproveVacation.Error(err));
+                    break;
+
+                case CalendarEventRemovedFromApprovers msg:
+                    this.ApproveCalendarEvent(msg.Event.EventId)
```

Comment:
sure about that?

Reply:
No, this is the only part, that wasn't completely refactored yet.

Reply:
will you change it in this PR?

Reply:
Sure, I'm working on it right now.

Reply:
Done in much better way.